File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
def solution(n): # 더블 포인터를 사용해보자
    answer = 0
    front, total, back = 1, 1, 2
    while front < back:
        if total < n:
            total += back
            back += 1
        elif n < total:
            total -= front
            front += 1
        else:
            answer += 1
            total -= front
            front += 1
    return answer
    

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.java
class Solution {
    public int solution(int n) {
        int a = 0;
        int b = 1;
        while (b * (b - 1) / 2 < n) {
            if ((n - b * (b - 1) / 2) % b == 0) {
                a++;
            }
            b++;
        }
        return a;
    }
}